### PR TITLE
Add script for generating steps, update codelab

### PR DESCRIPTION
**Pull request by collinjackson**
Updates codelab for flutter/flutter#3431

### CODE REVIEW

File: tool/generate_steps.sh [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+  git status
+  echo Error: You need a clean git working tree before running this command.
+  exit 1
+fi
+
+for cur in `seq 0 5`;
+do
+  echo Downloading step_$cur...
+  rm -rf step_$cur
+  git clone --depth=1 --branch=step_$cur git@github.com:collinjackson/codelab_steps.git step_$cur
```
Comment (abarth) [inline tool/generate_steps.sh]: Do you intend to reference `collinjackson` here or is there a version of this that's in the flutter github org?

**Comment by abarth** [comment]
LGTM except for the reference to your personal github.

File: step_1/lib/main.dart [hunk 2]
```diff
@@ -9,13 +9,11 @@ import 'package:flutter/material.dart';
 void main() {
   runApp(new MaterialApp(
     title: "Firechat",
+    home: new ChatScreen(),
```
Comment (Hixie) [inline step_1/lib/main.dart]: our style is to put children (like "home" here) last, fwiw

File: tool/generate_steps.sh (hunk 1 above)
Comment (collinjackson) [inline tool/generate_steps.sh]: I'll move it to the flutter github org.

File: step_1/lib/main.dart (hunk 2 above)
Comment (collinjackson) [inline step_1/lib/main.dart]: fixed

**Pull request merged**